core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -230,6 +230,66 @@ default boolean isComplete() {
      */
     void subscribe(Subscriber<? super T> subscriber, EventExecutor executor, boolean withPooledObjects);
 
+    /**
+     * Subscribes to this {@link StreamMessage} and retrieves all elements from it.
+     * The returned {@link CompletableFuture} will be completed exceptionally when one of the
+     * following exceptions is raised:
```

Comment:
The returned future will fail with other exceptions than the two exceptions below when the stream is terminated with other exceptions via `onError(Throwable)`, so I think we need to rephrase 'when ....' a little bit. How about this:

```
Subscribes to this {@link StreamMessage} and retrieves all elements from it.
The returned {@link CompletableFuture} may be completed exceptionally with the following exceptions:

<ul>
  <li>{@link IllegalStateException} if other {@link Subscriber} subscribed to this stream already.</li>
  <li>{@link AbortedStreamException} if this stream has been {@linkplain #abort() aborted}.</li>
  <li>Other exceptions that occurred due to an error while retrieving the elements.</li>
</ul>
```

Reply:
Got it. Will add the third condition to the Javadoc of `subscribe(...)` as well.